powersimdata/scenario/execute.py

```diff
@@ -189,54 +191,70 @@ def _launch_on_server(self, threads=None, extract_data=True):
             # Use the -t flag as defined in call.py in REISE.jl
             extra_args.append("--threads " + str(threads))
 
+        if solver:
+            extra_args.append("--solver", solver)
+
         if not isinstance(extract_data, bool):
             raise TypeError("extract_data must be a boolean: 'True' or 'False'")
         if extract_data:
             extra_args.append("--extract-data")
 
         return self._run_script("call.py", extra_args=extra_args)
 
-    def _launch_in_container(self, threads):
+    def _launch_in_container(self, threads, solver):
         """Launches simulation in container via http call
 
         :param int/None threads: the number of threads to be used. This defaults to None,
             where None means auto.
+        :param str solver: the solver used for optimization. This defaults to
+            None, which translates to gurobi
         :return: (*requests.Response*) -- the http response object
         """
         scenario_id = self._scenario_info["id"]
         url = f"http://{server_setup.SERVER_ADDRESS}:5000/launch/{scenario_id}"
-        resp = requests.post(url, params={"threads": threads})
+        resp = requests.post(url, params={"threads": threads, "solver": solver})
         if resp.status_code != 200:
             print(
                 f"Failed to launch simulation: status={resp.status_code}. See response for details"
             )
         return resp
 
-    def launch_simulation(self, threads=None, extract_data=True):
+    def _check_threads(self, threads):
+        if threads:
+            if not isinstance(threads, int):
+                raise TypeError("threads must be an int")
+            if threads < 1:
+                raise ValueError("threads must be a positive value")
+
+    def _check_solver(self, solver):
+        solvers = ("gurobi", "glpk")
+        if solver is not None and solver.lower() not in solvers:
+            raise ValueError(f"Invalid solver: options are {solvers}")
+
+    def launch_simulation(self, threads=None, extract_data=True, solver=None):
         """Launches simulation on target environment (server or container)
 
         :param int/None threads: the number of threads to be used. This defaults to None,
             where None means auto.
         :param bool extract_data: whether the results of the simulation engine should
             automatically extracted after the simulation has run. This defaults to True.
+        :param str solver: the solver used for optimization. This defaults to
+            None, which translates to gurobi
         :raises TypeError: if threads is not an int
-        :raises ValueError: if threads is not a positive value
+        :raises ValueError: if threads is not a positive value or invalid
+            solver provided
```

Comment:
You can remove all the exceptions in the docstring

Reply:
Yeah I know usually we limit the `raises` to the local exceptions, but felt like in this case it's clearer to keep it as if the `_check_*` methods are just an extension of this one (and it's unlikely the docstring on those would be useful, since I'd expect a user to do `help(scenario.state.launch_simulation)`. Kinda gray area though, what do you think? I can stick with the pattern if that's preferable.

Reply:
Not sure. If you do help(scenario.state.launch_simulation) you will see the expected type of the parameters through `:param str solver:` so you are not completely in the dark.

Reply:
That's true. Anyway, I updated it to be consistent since I think that's enough of a reason on its own.